### PR TITLE
JetBrains JECF aarch64 linux

### PR DESCRIPTION
**Pull request by sequencer**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by LunNova** [comment]
Result of `nixpkgs-review pr 250152` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>19 packages built:</summary>
  <ul>
    <li>greenfoot</li>
    <li>jetbrains.clion</li>
    <li>jetbrains.datagrip</li>
    <li>jetbrains.dataspell</li>
    <li>jetbrains.gateway</li>
    <li>jetbrains.goland</li>
    <li>jetbrains.idea-community</li>
    <li>jetbrains.idea-ultimate</li>
    <li>jetbrains.jcef</li>
    <li>jetbrains.jcef.unpacked</li>
    <li>jetbrains.jdk</li>
    <li>jetbrains.jdk.debug</li>
    <li>jetbrains.mps</li>
    <li>jetbrains.phpstorm</li>
    <li>jetbrains.pycharm-community</li>
    <li>jetbrains.pycharm-professional</li>
    <li>jetbrains.rider</li>
    <li>jetbrains.ruby-mine</li>
    <li>jetbrains.webstorm</li>
  </ul>
</details>

**Comment by LunNova** [comment]
Commits/PR title should have the attr path of the package as prefix, eg jetbrains.jdk:  or jetbrains.jcef: 

Tested on x86_64 and aarch64 boxes. aarch64 this gets jetbrains.jdk and jetbrains.jcef to build, didn't previously.

File: pkgs/development/compilers/jetbrains-jdk/jcef.nix [hunk 1]
```diff
@@ -73,6 +73,25 @@ let rpath = lib.makeLibraryPath [
...
+  };
+  "aarch64-linux" = {
+    clang_format_version = "dd736afb28430c9782750fc0fd5f0ed497399263";
+    clang_format_sha256 = "sha256-4H6FVO9jdZtxH40CSfS+4VESAHgYgYxfCBFSMHdT0hE=";
+    cef_platform = "linuxarm64";
+    cef_sha256 = "sha256-gCDIfWsysXE8lHn7H+YM3Jag+mdbWwTQpJf0GKdXEVs=";
+  };
+};
+
+cef_platform = lookupTable.${builtins.currentSystem}.cef_platform;
```
Comment (LunNova) [inline pkgs/development/compilers/jetbrains-jdk/jcef.nix:91]: Can't use `builtins.currentSystem` in pure contexts like flakes.

`stdenv.system` instead?

**Review by LunNova — CHANGES_REQUESTED** [review]

File: pkgs/development/compilers/jetbrains-jdk/jcef.nix [hunk 2]
```diff
@@ -73,6 +73,25 @@ let rpath = lib.makeLibraryPath [
 ];
 
 buildType = if debugBuild then "Debug" else "Release";
+lookupTable = {
+  "x86_64-linux" = {
+    clang_format_version = "dd736afb28430c9782750fc0fd5f0ed497399263";
```
Comment (LunNova) [inline pkgs/development/compilers/jetbrains-jdk/jcef.nix:78]: It looks like these  (clang_format_*) don't vary by system so didn't need moved.

**Comment by wegank** [comment]
Closed by #260943.

**Pull request closed without merging**